[CI] DataStreamGetWriteIndexTests testPickingBackingIndicesPredefinedDates failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/xyy3qgfryjs2y/tests/:modules:data-streams:test/org.elasticsearch.datastreams.DataStreamGetWriteIndexTests/testPickingBackingIndicesPredefinedDates

**Reproduction line:**
`./gradlew ':modules:data-streams:test' --tests "org.elasticsearch.datastreams.DataStreamGetWriteIndexTests.testPickingBackingIndicesPredefinedDates" -Dtests.seed=123E35694A81242C -Dtests.locale=da-DK -Dtests.timezone=Asia/Pontianak -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.datastreams.DataStreamGetWriteIndexTests&tests.test=testPickingBackingIndicesPredefinedDates

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: not null
     but: was null

  at __randomizedtesting.SeedInfo.seed([123E35694A81242C:DA947B7805F37533]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.datastreams.DataStreamGetWriteIndexTests.testPickingBackingIndicesPredefinedDates(DataStreamGetWriteIndexTests.java:93)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamGetWriteIndexTests.java
+++ b/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamGetWriteIndexTests.java
@@ -17,6 +17,7 @@ import org.elasticsearch.action.admin.indices.rollover.MetadataRolloverService;
 import org.elasticsearch.action.index.IndexRequest;
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.cluster.metadata.ComposableIndexTemplate;
+import org.elasticsearch.cluster.metadata.DataStream;
 import org.elasticsearch.cluster.metadata.DataStreamTestHelper;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.cluster.metadata.Metadata;
@@ -73,7 +74,6 @@ public class DataStreamGetWriteIndexTests extends ESTestCase {
     private MetadataRolloverService rolloverService;
     private MetadataCreateDataStreamService createDataStreamService;
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/85056")
     public void testPickingBackingIndicesPredefinedDates() throws Exception {
         Instant time = DateFormatters.from(MILLIS_FORMATTER.parse("2022-03-15T08:29:36.547Z")).toInstant();
 
@@ -90,10 +90,12 @@ public class DataStreamGetWriteIndexTests extends ESTestCase {
         var result = rolloverOver(state, "logs-myapp", time);
         state = result.clusterState();
 
-        backingIndex = state.getMetadata().index(".ds-logs-myapp-2022.03.15-000002");
+        DataStream dataStream = state.getMetadata().dataStreams().get("logs-myapp");
+        backingIndex = state.getMetadata().index(dataStream.getIndices().get(1));
         assertThat(backingIndex, notNullValue());
         assertThat(backingIndex.getSettings().get("index.time_series.start_time"), equalTo("2022-03-15T10:29:36.000Z"));
         assertThat(backingIndex.getSettings().get("index.time_series.end_time"), equalTo("2022-03-15T12:29:36.000Z"));
+        String secondBackingIndex = backingIndex.getIndex().getName();
 
         // first backing index:
         {
@@ -119,14 +121,14 @@ public class DataStreamGetWriteIndexTests extends ESTestCase {
             for (int i = 0; i < 256; i++) {
                 String timestamp = MILLIS_FORMATTER.formatMillis(randomLongBetween(start, end));
                 var writeIndex = getWriteIndex(state, "logs-myapp", timestamp);
-                assertThat(writeIndex.getName(), equalTo(".ds-logs-myapp-2022.03.15-000002"));
+                assertThat(writeIndex.getName(), equalTo(secondBackingIndex));
             }
         }
 
         // Borderline (again):
         {
             var writeIndex = getWriteIndex(state, "logs-myapp", "2022-03-15T12:29:35.999Z");
-            assertThat(writeIndex.getName(), equalTo(".ds-logs-myapp-2022.03.15-000002"));
+            assertThat(writeIndex.getName(), equalTo(secondBackingIndex));
         }
 
         // Outside the valid temporal ranges:
@@ -143,7 +145,6 @@ public class DataStreamGetWriteIndexTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/85056")
     public void testPickingBackingIndicesNanoTimestamp() throws Exception {
         Instant time = DateFormatters.from(NANOS_FORMATTER.parse("2022-03-15T08:29:36.123456789Z")).toInstant();
 
@@ -160,10 +161,12 @@ public class DataStreamGetWriteIndexTests extends ESTestCase {
         var result = rolloverOver(state, "logs-myapp", time);
         state = result.clusterState();
 
-        backingIndex = state.getMetadata().index(".ds-logs-myapp-2022.03.15-000002");
+        DataStream dataStream = state.getMetadata().dataStreams().get("logs-myapp");
+        backingIndex = state.getMetadata().index(dataStream.getIndices().get(1));
         assertThat(backingIndex, notNullValue());
         assertThat(backingIndex.getSettings().get("index.time_series.start_time"), equalTo("2022-03-15T10:29:36.000Z"));
         assertThat(backingIndex.getSettings().get("index.time_series.end_time"), equalTo("2022-03-15T12:29:36.000Z"));
+        String secondBackingIndex = backingIndex.getIndex().getName();
 
         // first backing index:
         {
@@ -189,14 +192,14 @@ public class DataStreamGetWriteIndexTests extends ESTestCase {
             for (int i = 0; i < 256; i++) {
                 String timestamp = NANOS_FORMATTER.formatMillis(randomLongBetween(start, end));
                 var writeIndex = getWriteIndex(state, "logs-myapp", timestamp);
-                assertThat(writeIndex.getName(), equalTo(".ds-logs-myapp-2022.03.15-000002"));
+                assertThat(writeIndex.getName(), equalTo(secondBackingIndex));
             }
         }
 
         // Borderline (again):
         {
             var writeIndex = getWriteIndex(state, "logs-myapp", "2022-03-15T12:29:35.999999999Z");
-            assertThat(writeIndex.getName(), equalTo(".ds-logs-myapp-2022.03.15-000002"));
+            assertThat(writeIndex.getName(), equalTo(secondBackingIndex));
         }
     }
 

